clients/api-entreprise/client.ts

```diff
@@ -20,14 +21,16 @@ export type IAPIEntrepriseResponse<T> = {
  */
 export default async function clientAPIEntreprise<T, U>(
   route: string,
-  mapToDomainObject: (e: T) => U,
-  recipientSiret?: string
+  mapToDomainObject: (e: T) => U
 ) {
-  if (!recipientSiret) {
+  const callerInfos = await sensitiveRequestCallerInfos();
```

Comment:
On introduit une dépendance implicite à un objet global. C'est commode, mais c'est dangereux : 
- Moins testable ;
- Couplage du client au framework (il faut pouvoir accéder à un objet global de contexte de requête, ce qui n'est pas donné dans tous les frameworks web) ;
- Potentiels effets de bord.

J'aurai préféré d'un point de vue conceptuel avoir cette logique dans la couche métier, via un décorateur au besoin, et que le `clientAPIEntreprise` attende toujours un email + siret en paramètre.

Mais : le code est plus clair et concis ainsi. Donc, ça ne me dérange pas plus que ça, c'est juste que c'est pas trop dans les canons d'archi.

Reply:
Totalement d'accord, j'ai pas mal hésité. 

In fine j'ai trouvé que cela valait le coup car le code qui en résulte est plus safe (tous les appels API Entreprise vont passer par ces checks)